src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -62,24 +65,46 @@ private HttpSystemProxy(WinInetProxyHelper proxyHelper, SafeWinHttpHandle sessio
                 {
                     // Process bypass list for manual setting.
                     string[] list = proxyHelper.ProxyBypass.Split(';');
-                    List<string> tmpList = new List<string>();
+                    _bypass = new List<Regex>(list.Length);
 
                     foreach (string value in list)
                     {
                         string tmp = value.Trim();
+                        if (tmp.Length == 0)
+                        {
+                            continue;
+                        }
                         if (tmp == "<local>")
                         {
                             _bypassLocal = true;
                             continue;
                         }
-                        if (tmp.Length > 0)
+                        try
+                        {
+                            // Escape any special characters and unescape * to get wildcard pattern match.
+                            Regex re = new Regex(Regex.Escape(tmp).Replace("\\*", ".*?") + "$",
+                                            RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+                            _bypass.Add(re);
+                        }
+                        catch
                         {
-                            tmpList.Add(tmp);
+                            if (NetEventSource.IsEnabled)
+                            {
+                                NetEventSource.Info(this, "Failed to process " + tmp + " from bypass list.");
```

Comment:
Nit: this can use string interpolation:
```C#
NetEventSource.Info(this, "Failed to process '{tmp}' from bypass list.");
```
It'd be good to put the string in quotes as well, as I've done.

Reply:
Ok.  We use interpolation in other places, as the tracing is guarded behind the IsEnabled switch, and there's already a lot of overhead when turning on tracing, so we've generally then opted for ease of reading/writing.  But what you have is fine.